### PR TITLE
Doc: Adds example of exploding lists into columns instead of storing in dataframe cells 

### LINKED ISSUE #17027: DOC: section on caveats of storing lists inside DataFrame/Series
xref to a lot of issues, for example #16864

I think we could use a doc section stating storing nested lists/arrays inside a pandas object is preferred to be avoided, showing the downsides (perf, memory use) and a worked out example of an alternative.  This seems to be earned knowledge that many have, but not sure we do a good job stating it clearly.

Closely related, might also benefit from a little section encouraging use of Python core data structures when appropriate.

probably goes here - http://pandas.pydata.org/pandas-docs/stable/gotchas.html

### PR DESCRIPTION
**Pull request by mgautam98**
- [x] closes #17027

This is the continuation of PR #19215

### CODE REVIEW

File: doc/source/gotchas.rst [hunk 1]
```diff
@@ -336,3 +336,94 @@ constructors using something similar to the following:
 See `the NumPy documentation on byte order
 <https://docs.scipy.org/doc/numpy/user/basics.byteswapping.html>`__ for more
 details.
+
+
+Alternative to storing lists in DataFrame Cells
```
Comment (WillAyd) [inline doc/source/gotchas.rst]: This isn't an "alternative" to lists as much as a way of just reshaping values; this is better phrased as just "Exploding List Items" or something to the effect

File: doc/source/gotchas.rst [hunk 2]
```diff
@@ -336,3 +336,94 @@ constructors using something similar to the following:
...
+-----------------------------------------------
+Storing nested lists/arrays inside a pandas object should be avoided for performance and memory use reasons. Instead they should be "exploded" into a flat ``DataFrame`` structure.
```
Comment (WillAyd) [inline doc/source/gotchas.rst]: Related to comment above this isn't an "alternative" to using lists within a DataFrame

File: doc/source/gotchas.rst [hunk 3]
```diff
@@ -336,3 +336,94 @@ constructors using something similar to the following:
...
+
+Example of exploding nested lists into a DataFrame:
+
+.. ipython:: python
+
+   df = pd.DataFrame({'name': ['A.J. Price'] * 3,
```
Comment (WillAyd) [inline doc/source/gotchas.rst]: I believe this is copied from the following SO article:

https://stackoverflow.com/questions/32468402/how-to-explode-a-list-inside-a-dataframe-cell-into-separate-rows

Need to be careful copy / pasting items from SO into the code base. Would have to get express permission from author to use

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: doc/source/gotchas.rst (hunk 3 above)
Comment (TomAugspurger) [reply doc/source/gotchas.rst]: I think SO code snippets are CC-By so as long as we link back to the source (which we should be doing anything) then we're good.

https://stackoverflow.com/help/licensing

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by datapythonista** [comment]
What I use to expand a `Series` with lists in the values to the corresponding `DataFrame` is:
```python
import pandas

genres = pandas.Series([['drama', 'romance'], ['romance'], ['comedy', 'action']])
genres.str.join(',').str.get_dummies(',')
```

So, I think the content of this PR shouldn't be in the documentation (I know you just used the existing PR @mgautam98, sorry I didn't see that earlier).

May be a short entry with that to the cookbook could be useful.

Closing, if anybody disagrees, please reopen.

**Pull request closed without merging**